database/scrapers/building_scrapper.ts

```diff
@@ -0,0 +1,158 @@
+      const buildingsData = buildingsBody.data;
+
+      const formattedBuildingData = await Promise.all(
+        buildingsData.map(async (data) => {
+          const addressArray = data.addres.split(",");
+          return {
+            id: data.id,
+            identifier: data.name,
+            specialName: data.naturalName,
+            iconType: BuildingIcon.Icon,
+            campusId: -1,
+            addressLine1: addressArray.shift(),
+            addressLine2: addressArray.shift(),
+            latitude: data.latitude,
+            longitude: data.longitude,
+            haveFood: data.food ?? false,
+            cover: await this.uploadAndGetKey(data),
+            externalDigitalGuideMode: data.externalDigitalGuideMode,
+            externalDigitalGuideIdOrURL: data.externalDigitalGuideIdOrURL,
+            createdAt: resolveDate(data.createdAt),
+            updatedAt: resolveDate(data.updatedAt),
+          };
+        }),
+      );
+      for (const buildingEntry of formattedBuildingData) {
+        const campus = campusesMap.get(buildingEntry.identifier);
+        if (campus === undefined) {
+          this.logger.error(
+            `No campus assigned to building ${buildingEntry.identifier}`,
+          );
+        } else {
+          // update campuses
+          campus.cover = buildingEntry.cover;
+          buildingEntry.campusId = campus.id;
+          await campus.save();
+        }
+      }
+      await Building.createMany(formattedBuildingData);
+      this.logger.info("Buildings created !");
+      task.update(`finished running ${this.name} scrapper`);
+    } catch (err) {
+      this.logger.error("error within buildings and campuses scrapper");
+      throw new Error(`${err}`);
+    }
+  }
+
+  private async uploadAndGetKey(data: BuildingsData): Promise<string> {
+    try {
+      const imageKey = data.cover;
+      if (imageKey === null) {
+        this.logger.warning(`no image for building: [${data.id}]`);
+        return "";
+      }
+      const response = await fetch(`${assetsPath}${imageKey}`);
+      if (!response.ok) {
+        throw new Error(response.statusText);
+      }
+      const body = response.body;
+      if (body === null) {
+        throw new Error(
+          `No file contents for ${imageKey} for building: [${data.id}]`,
+        );
+      }
+      let mediaType = "";
+      for (const [header, value] of response.headers.entries()) {
+        if (header === "content-type") {
+          if (value.startsWith("image")) {
+            mediaType = value.split("/").pop() ?? "";
+          }
+        }
+      }
+      return await this.filesService.uploadStream(
+        Readable.fromWeb(body),
+        mediaType,
+      );
+    } catch (err) {
+      this.logger.error(`failed to upload the files: ${err}`);
+      return "";
+    }
+  }
+}
+
+function resolveDate(dateString: string): DateTime<true> {
+  const date = DateTime.fromISO(dateString);
+  return date.isValid ? date : DateTime.local();
+}
```

Comment:
minor nitpick: `DateTime.local()` is the same as `DateTime.now()`, according to the docs.